cmd/chainbridge/deploycontracts.go

```diff
@@ -0,0 +1,271 @@
+
+	return auth, nil
+
+}
+
+func accountSetUp(port string, validators int, deployPK string) (*ethclient.Client, *bind.TransactOpts, common.Address, []common.Address, error) {
+
+	client, err := ethclient.Dial("http://localhost:" + port)
+	if err != nil {
+		log.Error(err.Error())
+		return nil, nil, ZERO_ADDRESS, nil, err
+	}
+
+	privateKey, err := crypto.HexToECDSA(deployPK)
+	if err != nil {
+		log.Error(err.Error())
+		return nil, nil, ZERO_ADDRESS, nil, err
+	}
+
+	publicKey := privateKey.Public()
+	publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+	if !ok {
+		log.Error(err.Error())
+		return nil, nil, ZERO_ADDRESS, nil, err
+	}
+
+	deployAddress := crypto.PubkeyToAddress(*publicKeyECDSA)
+	nonce, err := client.PendingNonceAt(context.Background(), deployAddress)
+	if err != nil {
+		log.Error(err.Error())
+		return nil, nil, ZERO_ADDRESS, nil, err
+	}
+
+	gasPrice, err := client.SuggestGasPrice(context.Background())
+	if err != nil {
+		log.Error(err.Error())
+		return nil, nil, ZERO_ADDRESS, nil, err
+	}
+
+	auth := bind.NewKeyedTransactor(privateKey)
+	auth.Nonce = big.NewInt(int64(nonce))
+	auth.Value = big.NewInt(0)
+	auth.GasLimit = uint64(6721975)
+	auth.GasPrice = gasPrice
+
+	validatorAddresses := createValidatorSlice(VALIDATOR_ADDRESS, validators)
+
+	return client, auth, deployAddress, validatorAddresses, nil
+
+}
+
+func deployReceiver(auth *bind.TransactOpts, client *ethclient.Client, validatorAddresses []common.Address, depositThreshold *big.Int, validatorThreshold *big.Int, deployAddress common.Address) (common.Address, error) {
+
+	auth, err := updateNonce(auth, client, deployAddress)
+	if err != nil {
+		log.Error("error getting most recent nonce!")
+		return ZERO_ADDRESS, err
+	}
+
+	recAddr, _, _, err := receiver.DeployReceiver(auth, client, validatorAddresses, depositThreshold, validatorThreshold)
+	if err != nil {
+		log.Error("error deploying reciever instance")
+		return ZERO_ADDRESS, err
+	}
+
+	return recAddr, nil
+}
+
+func deployEmitter(auth *bind.TransactOpts, client *ethclient.Client, deployAddress common.Address) (common.Address, error) {
+
+	auth, err := updateNonce(auth, client, deployAddress)
+	if err != nil {
+		log.Error("error getting most recent nonce!")
+		return ZERO_ADDRESS, err
+	}
+
+	emitterAddr, _, _, err := emitter.DeployEmitter(auth, client)
+	if err != nil {
+		log.Error("error deploying emitter instance")
+		return ZERO_ADDRESS, err
+	}
+
+	return emitterAddr, nil
+
+}
+
+func deploySimpleEmitter(auth *bind.TransactOpts, client *ethclient.Client, deployAddress common.Address) (common.Address, error) {
+
+	auth, err := updateNonce(auth, client, deployAddress)
+	if err != nil {
+		log.Error("error getting most recent nonce!")
+		return ZERO_ADDRESS, err
+	}
+
+	sEmitterAddr, _, _, err := simpleEmitter.DeploySimpleEmitter(auth, client)
+	if err != nil {
+		log.Error("error deploying simple emitter instance")
+		return ZERO_ADDRESS, err
+	}
+
+	return sEmitterAddr, nil
+}
+
```

Comment:
I think @GregTheGreek mentioned that we don't really use SimpleEmitter anywhere and that is should be removed, would this be a good ticket to remove it?

Reply:
Not a terrible idea. @steviezhang I would remove this instance of deploying the simple emitter. I'll open a separate issue.

Reply:
will do